jvector-examples/src/main/java/io/github/jbellis/jvector/example/util/Hdf5Loader.java

```diff
@@ -20,33 +20,49 @@
 import io.github.jbellis.jvector.vector.VectorUtil;
 import io.jhdf.HdfFile;
 
+import java.io.IOException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.*;
 
 public class Hdf5Loader {
     public static DataSet load(String pathStr) {
-        // infer the similarity
-        VectorSimilarityFunction similarityFunction;
-        if (pathStr.contains("angular")) {
+        final VectorSimilarityFunction similarityFunction;
+
+        boolean isHdf5 = pathStr.endsWith("hdf5");
+        if (isHdf5) {
+            if (pathStr.contains("angular")) {
+                similarityFunction = VectorSimilarityFunction.DOT_PRODUCT;
+            } else if (pathStr.contains("euclidean")) {
+                similarityFunction = VectorSimilarityFunction.EUCLIDEAN;
+            } else {
+                throw new IllegalArgumentException("Unknown similarity function -- expected angular or euclidean for " + pathStr);
+            }
+        } else {
+            // assuming sift
             similarityFunction = VectorSimilarityFunction.DOT_PRODUCT;
         }
-        else if (pathStr.contains("euclidean")) {
-            similarityFunction = VectorSimilarityFunction.EUCLIDEAN;
-        }
-        else {
-            throw new IllegalArgumentException("Unknown similarity function -- expected angular or euclidean for " + pathStr);
-        }
 
         // read the data
         float[][] baseVectors;
         float[][] queryVectors;
         int[][] groundTruth;
+
         Path path = Paths.get(pathStr);
-        try (HdfFile hdf = new HdfFile(path)) {
-            baseVectors = (float[][]) hdf.getDatasetByPath("train").getData();
-            queryVectors = (float[][]) hdf.getDatasetByPath("test").getData();
-            groundTruth = (int[][]) hdf.getDatasetByPath("neighbors").getData();
+        if (isHdf5) {
```

Comment:
don't do this, make the caller use SiftLoader where appropriate

Reply:
it looks like you did this to re-use the scrub code, but (1) the nytimes dataset in hdf5 format is the only one that we've needed to scrub, (2) if we do need to re-use it, let's move it out of Hdf5Loader (into DataSet perhaps)